### PR TITLE
feat: add mssql subs/tests

### PR DESCRIPTION
**Pull request by markwolff**
Requires https://github.com/microsoft/node-diagnostic-channel/pull/68
- Adds TDS patch subscribers, function tests

### CODE REVIEW

File: AutoCollection/diagnostic-channel/tedious.sub.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export const subscriber = (event: IStandardEvent<ITediousData>) => {
+    clients.forEach((client) => {
+        const q = event.data.query;
+        const sql = (q.preparable && q.preparable.text) || q.plan || q.text || "unknown query";
+        const success = !event.data.error;
+        const conn = `${event.data.database.host}:${event.data.database.port}`;
+        client.trackDependency({
+            target: conn,
+            data: sql,
+            name: sql,
```
Comment (xiao-lix) [inline AutoCollection/diagnostic-channel/tedious.sub.ts:19]: Is name same as data on purpose? Is it possible that data is long. Does name has character length limit or something?

File: AutoCollection/diagnostic-channel/tedious.sub.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export function enable(enabled: boolean, client: TelemetryClient) {
+    if (enabled) {
+        if (clients.length === 0) {
+            channel.subscribe<ITediousData>("tedious", subscriber);
+        };
+        clients.push(client);
+    } else {
+        clients = clients.filter((c) => c != client);
+        if (clients.length === 0) {
+            channel.unsubscribe("tedious", subscriber);
```
Comment (xiao-lix) [inline AutoCollection/diagnostic-channel/tedious.sub.ts:37]: Does unsubscribe not need `<ITediousData>` type?

File: Tests/FunctionalTests/RunFunctionalTests.js [hunk 3]
```diff
@@ -85,11 +86,13 @@ function cleanUpDocker() {
     run("docker stop ainjsmysql");
     run("docker stop ainjsredis");
     run("docker stop ainjspostgres");
+    // run("docker stop ainjsmssql");
```
Comment (xiao-lix) [inline Tests/FunctionalTests/RunFunctionalTests.js:89]: Are these commented on purpose?

**Review by xiao-lix — APPROVED** [review]

File: Tests/FunctionalTests/RunFunctionalTests.js (hunk 3 above)
Comment (markwolff) [reply Tests/FunctionalTests/RunFunctionalTests.js:89]: Yes it seems like mssql is not compatible with github runners. I don't want to block this PR on it though

File: AutoCollection/diagnostic-channel/tedious.sub.ts (hunk 1 above)
Comment (markwolff) [reply AutoCollection/diagnostic-channel/tedious.sub.ts:19]: Yes, the pattern for this repo seems to match the name with the data field for DB RDDs as the SQL command. e.g. the Mongo subscriber

https://github.com/microsoft/ApplicationInsights-node.js/blob/f7d6505e164fd5cfc42c6c0be35441fe76395de2/AutoCollection/diagnostic-channel/mongodb.sub.ts#L17-L22

All the sanitization/truncation for strings are handled at envelope creation. For example, here is where `name` is truncated:
https://github.com/microsoft/ApplicationInsights-node.js/blob/2855952b7348b407f370bef4a10af006425f6508/Library/EnvelopeFactory.ts#L118-L123

File: AutoCollection/diagnostic-channel/tedious.sub.ts (hunk 2 above)
Comment (markwolff) [reply AutoCollection/diagnostic-channel/tedious.sub.ts]: :+1: added it

**Comment by DevRCRun** [comment]
Any updates on when this might be complete please? After an initial test we are considering rolling app insights out across our estate, much of which is mssql. Thanks

**Pull request closed without merging**